src/coreclr/vm/eetwain.cpp

```diff
@@ -2124,7 +2128,167 @@ void EECodeManager::LeaveCatch(GCInfoToken gcInfoToken,
+    {
+        ULONG frameOffset = exInfo->m_frameIter.m_crawl.GetCodeInfo()->GetFrameOffsetFromUnwindInfo();
+        return pvRegDisplay->pCurrentContext->Rbp - 16 * frameOffset;
+    }
+    else
+    {
+        return pvRegDisplay->SP;
+    }
+#elif defined(HOST_ARM64)
+    return pvRegDisplay->SP;
+#elif defined(HOST_ARM)
+    return pvRegDisplay->SP;
+#elif defined(HOST_X86)
+    return pvRegDisplay->SP;
+#elif defined(HOST_RISCV64)
+    return pvRegDisplay->SP;
+#elif defined(HOST_LOONGARCH64)
+    return pvRegDisplay->SP;
+#endif
+}
+
+#endif // TARGET_WASM
+
+// Call catch, finally or filter funclet.
+// Return value:
+// * Catch funclet: address to resume at after the catch returns
+// * Finally funclet: unused
+// * Filter funclet: result of the filter funclet (EXCEPTION_CONTINUE_SEARCH (0) or EXCEPTION_EXECUTE_HANDLER (1))
+#ifndef USE_FUNCLET_CALL_HELPER
+// NOTE: This function must be prevented from calling the actual funclet via a tail call to ensure
+// that the m_csfEHClause is really set to what is a SP of the caller frame of the funclet. The
+// StackFrameIterator relies on this.
+#ifdef _MSC_VER
+#pragma optimize("", off)
+#elif defined(__clang__)
+[[clang::disable_tail_calls]]
+#else
+[[gnu::optimize("O0")]]
+#endif
+#endif // USE_FUNCLET_CALL_HELPER
+DWORD_PTR EECodeManager::CallFunclet(OBJECTREF throwable, void* pHandler, REGDISPLAY *pRD, ExInfo *pExInfo, bool isFilterFunclet)
+{
+    DWORD_PTR dwResult = 0;
+#ifdef TARGET_WASM
+    _ASSERTE(!"CallFunclet for WASM not implemented yet");
+#else
+    HandlerFn* pfnHandler = (HandlerFn*)pHandler;
+
+    pExInfo->m_csfEHClause = CallerStackFrame((UINT_PTR)GetCurrentSP());
+
+#ifdef USE_FUNCLET_CALL_HELPER
+    // Since the actual caller of the funclet is the assembly helper, pass the reference
+    // to the CallerStackFrame instance so that it can be updated.
+    UINT_PTR *pFuncletCallerSP = &(pExInfo->m_csfEHClause.SP);
+
+    if (isFilterFunclet)
+    {
+        // For invoking IL filter funclet, we pass the CallerSP to the funclet using which
+        // it will retrieve the framepointer for accessing the locals in the parent
+        // method.
+        dwResult = CallEHFilterFunclet(OBJECTREFToObject(throwable),
+#ifdef USE_CURRENT_CONTEXT_IN_FILTER
+                                       GetFrameRestoreBase(pRD->pCurrentContext),
+#else
+                                       GetFrameRestoreBase(pRD->pCallerContext),
+#endif
+                                       CastHandlerFn(pfnHandler),
+                                       pFuncletCallerSP);
+    }
+    else
+    {
+        dwResult = CallEHFunclet(OBJECTREFToObject(throwable),
+                                 CastHandlerFn(pfnHandler),
+                                 GetFirstNonVolatileRegisterAddress(pRD->pCurrentContext),
+                                 pFuncletCallerSP);
+    }
+#else
+    UINT_PTR establisherFrame = GetEstablisherFrame(pRD, pExInfo);
```

Comment:
```suggestion
    pExInfo->m_csfEHClause = CallerStackFrame((UINT_PTR)GetCurrentSP());
    
    UINT_PTR establisherFrame = GetEstablisherFrame(pRD, pExInfo);
```

Reply:
Addressing my remaining suggestion

Reply:
I am not sure what this suggestion is - the diff shows it in a very confusing way.

Reply:
The suggestion is to move `pExInfo->m_csfEHClause = CallerStackFrame((UINT_PTR)GetCurrentSP());` under `!USE_FUNCLET_CALL_HELPER`

More context: https://github.com/dotnet/runtime/pull/114180#discussion_r2029400483

Reply:
No, this needs to be done in both cases

Reply:
Hmm, actually, it doesn't. My mistake.